src/build/sw-precache.ts

```diff
@@ -10,40 +10,161 @@
+  cacheId?: string,
+  directoryIndex?: string;
+  dynamicUrlToDependencies?: {
+    [property: string]: string[]
+  };
+  handleFetch?: boolean;
+  ignoreUrlParametersMatching?: RegExp[];
+  importScripts?: string[];
+  logger?: Function;
+  maximumFileSizeToCacheInBytes?: number;
+  navigateFallback?: string;
+  navigateFallbackWhitelist?: RegExp[];
+  replacePrefix?: string;
+  runtimeCaching?: {
+    urlPattern: RegExp;
+    handler: string;
+    options?: {
+      cache: {
+        maxEntries: number;
+        name: string;
+      };
+    };
+  }[];
+  staticFileGlobs?: string[];
+  stripPrefix?: string;
+  templateFilePath?: string;
+  verbose?: boolean;
+}
+
+function generateServiceWorker(
     root: string,
     main: string,
-    deps: string[]): Promise<string> {
+    deps: string[],
+    buildRoot: string,
+    swConfig?: SWConfig
+  ): Promise<string> {
+  swConfig = swConfig || <SWConfig>{};
   deps = deps.map((p) => {
     console.assert(p.startsWith(root));
     return p.substring(root.length);
   });
   let mainHtml = main.substring(root.length);
-  let precacheFiles = new Set<string>();
+  let precacheFiles = new Set(swConfig.staticFileGlobs);
   deps.forEach((p) => precacheFiles.add(p));
   precacheFiles.add(mainHtml);
-  // TODO: make this more configurable
-  precacheFiles.add('/manifest.json');
-  precacheFiles.add('/bower_components/webcomponentsjs/webcomponents-lite.min.js');
-  precacheFiles.add('/images/*'); //TODO: pull from user precache config
 
-  //TODO: pull from user precache config
-  let buildRoot = 'build/bundled';
-
-  let precacheList = Array.from(precacheFiles.values());
+  let precacheList = Array.from(precacheFiles);
   precacheList = precacheList.map((p) => path.join(buildRoot, p));
-  console.log('precacheList', precacheList);
-
-  return swPrecache.generate({
-    staticFileGlobs: precacheList,
-    navigateFallback: mainHtml,
-    //TODO: pull from user precache config
-    navigateFallbackWhitelist: [/\/data\/.*json/],
-    stripPrefix: buildRoot,
-    // verbose: true,
-  });
+  console.log('precacheList' , precacheList);
+
+  swConfig.stripPrefix = buildRoot;
+  swConfig.staticFileGlobs = precacheList;
+  // TODO: determine if this is a good default, or pull from user precache config
+  swConfig.navigateFallback = mainHtml;
+
+  return swPrecache.generate(swConfig);
+}
+
+export interface SWPreCacheTransformOptions {
+  /**
+   * folder containing files to be served by the service worker.
+   */
+  root: string;
+  /**
+   * Main file to serve from service worker.
+   */
+  main: string;
+  /**
+   * Output folder for the service worker bundle
+   */
+  buildRoot: string;
+  /**
+   * Name of the output service worker file.
+   */
+  serviceWorkerName: string;
+  /**
+   * Promise that returns the list of files to be cached by the service worker.
+   *
+   * If not given, all files streamed to SWPreCacheTransform will be put into the precache list, except for the file matching `configFileName`.
```

Comment:
I'm not sure about that default behavior. The way we source files here that would be _everything_ including demos, tests, READMEs, etc. Since we won't not specify the files, would it be simpler to drop this?